Atomic/atomic.py

```diff
@@ -254,6 +273,54 @@ def push(self):
 
                 prevstatus = status
 
+
+    def _pull_dockertar_layers(self, repo, imagebranch, temp_dir, input_layers):
+        layers = {}
+        next_layer = {}
+        top_layer = None
+        for i in input_layers:
+            layer = i.replace("/layer.tar", "")
+            layers[layer] = os.path.join(temp_dir, i)
+            with open(os.path.join(temp_dir, layer, "json"), 'r') as f:
+                json_layer = json.loads(f.read())
+                parent = json_layer.get("parent")
+                if not parent:
+                    top_layer = layer
+                next_layer[parent] = layer
+
+        layers_map = {}
+        enc = sys.getdefaultencoding()
+        for k, v in layers.items():
+            out = subprocess.check_output([ATOMIC_LIBEXEC + '/dockertar-sha256-helper',
```

Comment:
so that given a layer in a docker save'd tarball we can compute the same checksum it has on the docker registry (the checksum found in the registry manifest is computed after the .tar file is gzipped).  Other gzip implementations generate a different file, even though the original .tar is the same.

It is the issue discussed here: https://github.com/projectatomic/skopeo/issues/11